code/game/objects/items/storage/storage.dm

```diff
@@ -719,4 +746,39 @@
 	var/obj/item/drawn_item = contents[length(contents)]
 	drawn_item.attack_hand(user)
 
+///Called during Initialize this will iterate over spawns_with and spawns_prob to create our initial contents
 /obj/item/storage/proc/PopulateContents()
+	if(LAZYLEN(spawns_with))
+		var/iterations = 0
+		while(iterations < spawns_mult)
+			iterations += 1
+			for(var/item in spawns_with)
+				new item(src)
+
+	if(LAZYLEN(spawns_prob) && spawns_prob_max)
+		var/list/spawnlists = list()
+		var/list/spawnprobs = list()
+		var/index = 1
+		for(var/list/spawnlist in spawns_prob)
+			var/listprob = spawnlist[1]
+			if(prob(listprob))
+				spawnlist -= listprob
+				spawnlists.len += 1
+				spawnlists[index] = spawnlist
+				spawnprobs += listprob
+				index += 1
```

Comment:
what is all of this doing?

Can you make it a bit clearer, with better var names / comment in the code?